app/models/project_metadata.rb

```diff
@@ -130,39 +125,60 @@ def user_list_params(counter, key_prefix)
         users.compact.uniq
       end
 
-      def project_timestamps
-        timestamps = {}
-        if project.metadata[:created_by].nil?
-          timestamps[:created_by] = current_user.uid
-          timestamps[:created_on] = Time.current.in_time_zone("America/New_York").iso8601
-
-        else
-          timestamps[:created_by] = project.metadata[:created_by]
-          timestamps[:created_on] = project.metadata[:created_on]
-          timestamps[:updated_by] = current_user.uid
-          timestamps[:updated_on] = Time.current.in_time_zone("America/New_York").iso8601
+      # Initializes values that we have defaults for.
+      def set_defaults
+        if @storage_capacity.nil?
+          @storage_capacity = {
+            size: { requested: 500, approved: nil },
+            unit: { requested: "GB", approved: nil }
+          }
         end
-        timestamps
+
+        if @storage_performance_expectations.nil?
+          @storage_performance_expectations = { requested: "Standard", approved: nil }
+        end
+
+        @project_purpose = "Research" if @project_purpose.nil?
       end
 
-      def form_metadata
-        ro_users = user_list_params(read_only_counter, "ro_user_")
-        rw_users = user_list_params(read_write_counter, "rw_user_")
-        data_users = {
-          data_user_read_only: ro_users,
-          data_user_read_write: rw_users
+      # Sets a value in the object if the value exists in the params
+      def set_value(params, key)
```

Comment:
Is this going to stop people from blanking out values in the future?  Aka if the param value is empty I can not set it to blank.

Reply:
Good catch. Fixed.